packages/fxa-auth-server/lib/routes/totp.js

```diff
@@ -76,6 +76,205 @@ module.exports = (
     ? `${config.serviceName} - ${environment}`
     : `${config.serviceName}`;
 
+  // Shared handlers for TOTP replace flows (used by legacy and /mfa routes)
+  function isMfaRoute(request) {
+    const path = request?.route?.path || '';
+    return typeof path === 'string' && path.startsWith('/mfa/');
+  }
+  async function handleTotpReplaceStart(request) {
+    log.begin('totp.replace.create', request);
+
+    const { uid } = request.auth.credentials;
+    const account = await db.account(uid);
+
+    // For legacy sessionToken routes, require a verified session
+    if (!isMfaRoute(request)) {
```

Comment:
Just a nit, but it _should_ be fine to leave this in for MFA routes too. The mfa auth strategy back fills the `auth.credentials` with the same meta data that the `sessionToken` strategy does. If this isn't working, the fix should be over in the auth strategy and not necessarily here. See this [PR](https://github.com/mozilla/fxa/pull/19415) for more context.